perfkitbenchmarker/pkb.py

```diff
@@ -195,35 +257,55 @@ def RunBenchmark(benchmark, collector, sequence_number, total_benchmarks):
         logging.exception('Prerequisite check failed for %s', benchmark_name)
         raise
 
-    start_time = time.time()
+    # Initialize timed interval objects.
+    TimedInterval = timing_util.TimedInterval
+    end_to_end_interval = TimedInterval('End to End')
+    resource_provisioning_interval = TimedInterval('Resource Provisioning')
+    bm_prepare_interval = TimedInterval('Benchmark Prepare')
+    bm_run_interval = TimedInterval('Benchmark Run')
+    bm_cleanup_interval = TimedInterval('Benchmark Cleanup')
+    resource_teardown_interval = TimedInterval('Resource Teardown')
```

Comment:
This is a big diff to enable timing. 

I wonder if this would be cleaner with an `IntervalTimer`, with a `Measure(self, interval_name)` method which could be invoked multiple times. End to end might need special treatment, but the rest can be handled identically, and you'd only need to pass a single extra arg to `DoPrepare`, etc.